File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Blessed.java
package com.watabou.pixeldungeon.actors.buffs;

import com.watabou.pixeldungeon.actors.Char;
import com.watabou.pixeldungeon.ui.BuffIndicator;

public class Blessed extends FlavourBuff {

	@Override
	public int icon() {
		return BuffIndicator.BLEESSED;
	}

	@Override
	public int defenceSkillBonus(Char chr) {
		return level();
	}

	@Override
	public int attackSkillBonus(Char chr) {
		return level();
	}
}

File: scripts/buffs/Blessed.lua
---
--- Blessed: attack and defence skill bonus (was actors/buffs/Blessed.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

return buff.init{
    desc  = function ()
        return {
            icon          = 42, -- BuffIndicator.BLEESSED
            name          = "BlessedBuff_Name",
            info          = "BlessedBuff_Info",
        }
    end,

    defenceSkillBonus = function(self, buff)
        return buff:level()
    end,

    attackSkillBonus = function(self, buff)
        return buff:level()
    end
}
